consent-enforcement/consent-enforcement-mediator/src/main/java/org/wso2/financial/services/apim/mediation/policies/consent/enforcement/utils/ConsentEnforcementUtils.java

```diff
@@ -0,0 +1,301 @@
+                JSONObject jwtClaims = decodeBase64(authHeader.split("\\.")[1]);
+
+                String consentIdClaimName = (String) configs.get(FinancialServicesConstants.CONSENT_ID_CLAIM_NAME);
+
+                if (!jwtClaims.isNull(consentIdClaimName) && !jwtClaims.getString(consentIdClaimName).isEmpty()) {
+                    consentIdClaim = jwtClaims.getString(consentIdClaimName);
+                }
+            }
+            return consentIdClaim;
+        }
+        return null;
+    }
+
+    /**
+     * Method to create the validation request payload.
+     *
+     * @param axis2MessageContext Axis2 message context
+     * @param requestHeaders Transport headers from the Axis2 message context
+     * @param additionalParams Additional parameters to be included in the request payload
+     * @return JSONObject representing the validation request payload
+     */
+    public static JSONObject createValidationRequestPayload(org.apache.axis2.context.MessageContext axis2MessageContext,
+                                                            Map<String, Object> requestHeaders,
+                                                            Map<String, Object> additionalParams) throws JSONException {
+
+        JSONObject validationRequest = new JSONObject();
+        JSONObject headers = new JSONObject();
+
+        requestHeaders.forEach(headers::put);
+        validationRequest.put(ConsentEnforcementConstants.HEADERS_TAG, headers);
+
+        JSONObject requestPayload = new JSONObject(JsonUtil.jsonPayloadToString(axis2MessageContext));
+        validationRequest.put(ConsentEnforcementConstants.BODY_TAG, requestPayload);
+
+        additionalParams.forEach(validationRequest::put);
+        return validationRequest;
+    }
+
+    /**
+     * Method to generate JWT.
+     * @param payload Payload to be signed
+     * @return Signed JWT
+     */
+    public static String generateJWT(String payload) {
+
+        return Jwts.builder()
+                .setPayload(payload)
+                .signWith(SignatureAlgorithm.RS512, getSigningKey())
+                .compact();
+    }
+
+    /**
+     * Method to invoke consent validation service when the JWT payload is provided.
+     *
+     * @param enforcementJWTPayload JWT Payload
+     * @return Response as a String
+     * @throws IOException When failed to invoke the validation endpoint or failed to parse the response.
+     */
+    public static String invokeConsentValidationService(String enforcementJWTPayload) throws IOException,
+            FinancialServicesException {
+
+        HttpPost httpPost = new HttpPost(getValidationEndpoint());
+        StringEntity params;
+        params = new StringEntity(enforcementJWTPayload);
+        httpPost.setEntity(params);
+        httpPost.setHeader(ConsentEnforcementConstants.CONTENT_TYPE_TAG, ConsentEnforcementConstants.JWT_CONTENT_TYPE);
+
+        String userName = configService.getAPIManagerConfiguration()
```

Comment:
Isn't these has to be properties of the mediator ,not as configs? which need to connect with consent endpoint?

Reply:
This is the key manager user name and password configured in the APIM toml to add as a Auth header when sending the consent validate request. Since the validate is deployed in the IS I feel its ok to get it from the APIM toml